channel/channel_test.go

```diff
@@ -119,6 +119,93 @@ func TestChannel(t *testing.T) {
 			t.Errorf("TestCrank: side effects mismatch (-want +got):\n%s", diff)
 		}
 	}
+	alicePrivateKey := common.Hex2Bytes(`caab404f975b4620747174a75f08d98b4e5a7053b691b41bcfc0d839d48b7634`)
+	bobPrivateKey := common.Hex2Bytes(`62ecd49c4ccb41a70ad46532aed63cf815de15864bc415c87d507afd6a5e8da2`)
+	testAddSignedState := func(t *testing.T) {
+		myC, _ := New(s, 0)
+		ss := state.NewSignedState(s)
+		sigA, err := ss.State().Sign(alicePrivateKey)
+		if err != nil {
+			t.Error(err)
+		}
+		sigB, err := ss.State().Sign(bobPrivateKey)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ss.AddSignature(sigA)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ss.AddSignature(sigB)
+		if err != nil {
+			t.Error(err)
+		}
+		if ok := myC.AddSignedState(ss); !ok {
+			t.Error("AddSignedState returned false")
+		}
+
+		// It should properly update the latestSupportedStateNum
+		if myC.latestSupportedStateTurnNum != 0 {
+			t.Errorf("Expected latestSupportedStateTurnNum of 0 but got %d", myC.latestSupportedStateTurnNum)
+
+		}
+		// verify the signatures
+		expectedSigs := []state.Signature{sigA, sigB}
+		for i := range myC.Participants {
+			gotSig, err := myC.SignedStateForTurnNum[s.TurnNum].GetParticipantSignature(uint(i))
+			if err != nil {
+				panic(err)
+			}
+			wantSig := expectedSigs[i]
+			if !gotSig.Equal(wantSig) {
+				t.Errorf("Expected to find signature %x at index 0, but got %x", wantSig, gotSig)
+			}
+		}
+	}
+	testAddSignedStates := func(t *testing.T) {
+		myC, _ := New(s, 0)
```

Comment:
Most of these tests share the same `channel` which means one test can easily break another. This is similar to #226

Reply:
Shall we make an issue for this?

Reply:
I've updated #226 to track this.